### DISCUSSION TITLE
Problem with Model table name

### DISCUSSION THREAD

**Discussion by mastermunj** [adonisjs/core#1601] [Help] [answered]
Hi,

I am trying to create a model named `BaseIdModel`, which all the models in my app will extend. The reason is that I want to keep id generation and createdAt, updatedAt columns definition common. Moreover, any common functionality I'd wish all models to have will be provided through this `BaseIdModel`.

When I try to query the database with any model that extends `BaseIdModel`, the table name used in the query is `base_id_model`.

##### app/Models/BaseIdModel.ts
```ts
import { DateTime } from 'luxon';
import { BaseModel, beforeSave, column } from '@ioc:Adonis/Lucid/Orm';
import Utils from 'App/Utils/Utils';

export default class BaseIdModel extends BaseModel {
  @column({ isPrimary: true })
  public id: number;

  @column.dateTime({ autoCreate: true })
  public createdAt: DateTime;

  @column.dateTime({ autoCreate: true, autoUpdate: true })
  public updatedAt: DateTime;

  @beforeSave()
  public static async generateId(model: BaseIdModel): Promise<void> {
    model.id = Utils.uniqueId();
  }
}
```

##### app/Models/Offer.ts
```ts
import { column } from '@ioc:Adonis/Lucid/Orm';
import BaseIdModel from './BaseIdModel';

export default class Offer extends BaseIdModel {
  @column()
  public name: string;

  @column()
  public active: boolean;
}
```

Whenever I call
```ts
Offer.create({name: 'Offer1'});
```
the query that gets generated is:
```sql
insert into "base_id_models" ("created_at", "id", "name", "updated_at") values ($1, $2, $3, $4)
```
with error

```sql
relation "base_id_models" does not exist
```

Kindly help.

**Comment by thetutlage** [comment, marked as answer]
Yeah, the inhertance story with models is not that great. Something we will fix in the next release. For now you will have to explicitly define the table name on the models.

```ts
import { column } from '@ioc:Adonis/Lucid/Orm';
import BaseIdModel from './BaseIdModel';

export default class Offer extends BaseIdModel {
   public static table = 'offers'
}
```

**Reply by mastermunj** [reply]
Got it, will give explicit names in model as of now.

Is there a way to get notified once the fix gets released?

**Reply by thetutlage** [reply]
Yes. we do annouce releases on twitter https://twitter.com/adonisframework. Otherwise you need to keep an eye on the releases page https://preview.adonisjs.com/releases/core/preview-rc-5/

Also, we have 6 weeks release schedule. So next release is due in October
